symengine/cwrapper.cpp

```diff
@@ -562,6 +565,194 @@ size_t vecbasic_size(CVecBasic *self)
     return self->m.size();
 }
 
+// C wrapper for Matrix
+
+struct CDenseMatrix {
+    SymEngine::DenseMatrix m;
+};
+
+struct CSparseMatrix {
+    SymEngine::CSRMatrix m;
+};
+
+CDenseMatrix *dense_matrix_new()
+{
+    return new CDenseMatrix;
+}
+
+CSparseMatrix *sparse_matrix_new()
+{
+    return new CSparseMatrix;
+}
+
+void dense_matrix_free(CDenseMatrix *self)
+{
+    delete self;
+}
+
+void sparse_matrix_free(CSparseMatrix *self)
+{
+    delete self;
+}
+
+void dense_matrix_init(CDenseMatrix *s)
```

Comment:
wrapping `DenseMatrix::DenseMatrix(const DenseMatrix &x)` this constructor.
using it for: `SymEngine::DenseMatrix(SymEngine::DenseMatrix)` in the ruby wrapper